mysql-test/suite/galera/t/MDEV-36116.test

```diff
@@ -0,0 +1,36 @@
+#
+# MDEV-36116: TOI crashes in debug assert if executing thread is killed.
+#
+
+--source include/galera_cluster.inc
+--source include/have_debug_sync.inc
+--source include/have_debug.inc
+
+--connect con1,127.0.0.1,root,,test,$NODE_MYPORT_1
+
+# Start TOI operation and wait for the thread to be killed.
+--connection node_1
+CALL mtr.add_suppression("CREATE TABLE isolation failure");
+
+--let $connection_id = `SELECT CONNECTION_ID()`
+SET DEBUG_SYNC = 'wsrep_kill_thd_before_enter_toi SIGNAL may_kill WAIT_FOR continue';
+--send
+  CREATE TABLE t1 (a INT) ENGINE=InnoDB;
+
+# Kill the thread and let it continue.
+--connection con1
+SET DEBUG_SYNC = 'now WAIT_FOR may_kill';
+--disable_query_log
+--eval KILL CONNECTION $connection_id
+--enable_query_log
+SET DEBUG_SYNC = 'now SIGNAL continue';
+
+--connection node_1
+--error 2013
+--reap
+
```

Comment:
Verify here that the table was not created on either nodes.

Reply:
Done.